### PR TITLE
feat: allow optional additionalStaticPaths function

### PR DESCRIPTION
**Pull request by orinokai**
This PR allows the `additionalStaticPaths` param to optionally be a function that receives the `resolvedConfig` object.

This allows the consuming code to use root/base file paths from the Vite config.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import glob from 'fast-glob'
 export interface NetlifyEdgePluginOptions {
   generateStaticManifest?: boolean
   generateEdgeFunctionsManifest?: boolean
-  additionalStaticPaths?: Array<string>
+  additionalStaticPaths?: Array<string> | ((config: ResolvedConfig) => string[])
```
Comment (ascorbic) [inline src/index.ts:10]: [pebble] could we be consistent with `Array<string>` syntax here?

Comment (nickytonline) [reply src/index.ts:10]: Might be worth an eslint rule if this is not a one off. https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/array-type.md

Comment (ascorbic) [reply src/index.ts:10]: Agreed, though in this case it's mostly about consistency

**Review by ericapisani — COMMENTED** [review]
These changes are looking great!

Is it possible to also add some test coverage for this functionality as well?

**Review by ascorbic — APPROVED** [review]
This package doesn't have any test coverage yet, so I don't think that should be a blocker. It can go in a follow-up that adds tests

**Comment by orinokai** [comment]
Closing because this was superseded by another feature

**Pull request closed without merging**